src/main/scala/nl.knaw.dans.easy.deposit/AgreementGenerator.scala

```diff
@@ -24,29 +23,26 @@ import nl.knaw.dans.easy.deposit.docs.AgreementData
 import nl.knaw.dans.easy.deposit.docs.JsonUtil._
 import nl.knaw.dans.lib.logging.DebugEnhancedLogging
 import org.eclipse.jetty.http.HttpStatus.OK_200
-import scalaj.http.BaseHttp
+import scalaj.http.{ BaseHttp, HttpResponse }
 
-import scala.io.Source
-import scala.util.{ Failure, Success, Try }
+import scala.util.Try
 
 case class AgreementGenerator(http: BaseHttp, url: URL, acceptHeader: String, connectionTimeoutMs: Int = 2000, readTimeoutMs: Int = 30000) extends DebugEnhancedLogging {
-  def generate(agreementData: AgreementData, id: UUID): Try[Array[Byte]] = {
+  def generate(agreementData: AgreementData, id: UUID): Try[Array[Byte]] = Try {
     val json = toJson(agreementData)
-    logger.info(s"calling easy-deposit-agreement-generator for $id with body: $json")
-    Try(http(url.toString).timeout(connectionTimeoutMs, readTimeoutMs).postData(json)
-        .header("content-type", "application/json")
-        .header("accept", acceptHeader)
-        .exec {
-      case (OK_200, _, is) =>
-        return Success(readAll(is))
-      case (_, _, is) =>
-        Source.fromInputStream(is).mkString
-    }).flatMap{ response =>
-      Failure(GeneratorError(s"Could not generate agreement for dataset $id", response))
-    }
-  }
+    logger.info(s"[$id] calling easy-deposit-agreement-generator with body $json")
 
-  private def readAll(is: InputStream) = {
-    Stream.continually(is.read).takeWhile(_ != -1).map(_.toByte).toArray
+    http(url.toString).timeout(connectionTimeoutMs, readTimeoutMs).postData(json)
+      .header("content-type", "application/json")
+      .header("accept", acceptHeader)
+      .asBytes match {
```

Comment:
Weird name chosen by this this library. It might as well represent the request rather than the response.